scripts/build-windows.ps1

```diff
@@ -0,0 +1,175 @@
+        $tempDownloadFile = New-TemporaryFile
+        $cmakeMsiFile = "$tempDownloadFile.cmake-3.17.3-win64-x64.msi"
+        Write-Output "Downloading cmake (temporary file location $cmakeMsiFile)"
+        Write-Output "Note: select the option to add cmake to path for this script to operate"
+        $client.DownloadFile("https://github.com/Kitware/CMake/releases/download/v3.17.3/cmake-3.17.3-win64-x64.msi", "$cmakeMsiFile")
+        Start-Process $cmakeMsiFile -Wait
+        Remove-Item $cmakeMsiFile
+        Write-Output "Cmake should have installed, this script will now exit because of path updates - please now re-run this script"
+        exit
+    }
+    else{
+        Write-Output "Quitting because cmake is required"     
+        exit
+    }
+}
+
+if ( $CXX11 -eq "OFF" ) {
+    # get pthreads (this is legacy, and is only availble in nuget for VS2015 and VS2013)
+    if ( $VS_VERSION -gt 2015 ) { 
+        Write-Output "Pthreads is not recommended for use beyond VS2015 and is not supported by this build script - aborting build"
+        exit
+    }
+    if ( $DEVENV_PLATFORM -eq 'Win32' ) { 
+        nuget install cinegy.pthreads-win32-$VS_VERSION -version 2.9.1.24 -OutputDirectory ../_packages
+    }
+    else {        
+        nuget install cinegy.pthreads-win64-$VS_VERSION -version 2.9.1.24 -OutputDirectory ../_packages
+    }
+}
+
+if ($STATIC_LINK_SSL -eq "ON") {
+    # requesting a static link will implicitly enable encryption support
+    Write-Output "Static linking to OpenSSL requested, will force encryption feature ON"
+    $ENABLE_ENCRYPTION = "ON"
+}
+
+# build the cmake command flags from arguments
+$cmakeFlags = "-DCMAKE_BUILD_TYPE=$CONFIGURATION " + 
+                "-DENABLE_STDCXX_SYNC=$CXX11 " + 
+                "-DENABLE_APPS=$BUILD_APPS " + 
+                "-DENABLE_ENCRYPTION=$ENABLE_ENCRYPTION " +
+                "-DOPENSSL_USE_STATIC_LIBS=$STATIC_LINK_SSL " + 
+                "-DENABLE_UNITTESTS=$UNIT_TESTS"
+
+# cmake uses a flag for architecture from vs2019, so add that as a suffix
+if ( $VS_VERSION -eq '2019' ) {    
+    $cmakeFlags += " -A `"$DEVENV_PLATFORM`""
+}
+
+# fire cmake to build project files
+$execVar = "cmake ../ -G`"$CMAKE_GENERATOR`" $cmakeFlags"
+Write-Output $execVar
+Invoke-Expression "& $execVar"
+
+# check build ran OK, exit if cmake failed
+if($LASTEXITCODE -ne 0) {
+    return $LASTEXITCODE
+}
+
+# run the set-version-metadata script to inject build numbers into appveyors console and the resulting DLL
+. $PSScriptRoot/set-version-metadata.ps1
+
+# execute msbuild to perform compilation and linking - if outside of appveyor, use vswhere to find msbuild
+if($Env:APPVEYOR) { 
+    msbuild SRT.sln /p:Configuration=$CONFIGURATION /p:Platform=$DEVENV_PLATFORM
+}
+else {
+    # check vswhere has been added to path, or is in the well-known location (true if VS2017 Update 2 or later installed)
+    if ($null -eq (Get-Command "vswhere.exe" -ErrorAction SilentlyContinue)) { 
```

Comment:
It should be done this way:
1. Check if `vswhere` is executable -> if so, save into $vswhereCmd
2. If not, check if `vswhere` in visual studio path is executable -> if so, save into $vswhereCmd
3. If not, exit with error information
4. Run `$vswhereCmd -version $MSBUILDVER -requires Microsoft.Component.MSBuild -find MSBuild\**\Bin\MSBuild.exe | select-object -first 1`

Reply:
Bad case of 'Iwantedtogohomeitis' - rewritten to be much less crazy.

Reply:
No problem - it's always the case that at first you write something that works, then something to look good :)